### ISSUE TITLE
jobs: TestRegistryGC is flaky

### ISSUE THREAD

**Issue by rafiss** [cockroachdb/cockroach#116702]
Seen here on a 23.2 PR: https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_BazelEssentialCi/13205327?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildChangesSection=true&expandBuildProblemsSection=true&expandBuildTestsSection=true

```
Failed
=== RUN   TestRegistryGC
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/426571b820df304bf0b1de52f751939f/logTestRegistryGC656873582
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:146: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    conditional_wrap.go:190:
        pkg/jobs/registry_test.go:246: (TestRegistryGC)
          NOTICE: .JobRegistry() called via implicit interface ApplicationLayerInterface;
        HINT: consider using .ApplicationLayer().JobRegistry() instead.
    conditional_wrap.go:194: TIP: consider replacing the test server initialization from:
            ts, ... := serverutils.StartServer(t, ...)
            defer ts.Stopper().Stop(...)
        to:
            srv, ... := serverutils.StartServer(t, ...)
            defer srv.Stopper().Stop(...)
            ts := srv.ApplicationLayer()
        See also: https://go.crdb.dev/p/testserver-and-cluster-virtualization
    registry_test.go:247: TransactionRetryWithProtoRefreshError: ReadWithinUncertaintyIntervalError: read at time 1702923744.592794773,0 encountered previous write with future timestamp 1702923744.592794773,1 (local=1702923744.588748153,1) within uncertainty interval `t <= (local=1702923744.592794773,0, global=1702923745.092794773,0)`; observed timestamps: [{1 1702923744.592794773,0}]: "sql txn" meta={id=62d42348 key=/Min iso=Serializable pri=0.01132670 epo=0 ts=1702923744.592794773,0 min=1702923744.592794773,0 seq=0} lock=false stat=PENDING rts=1702923744.592794773,0 wto=false gul=1702923745.092794773,0
    panic.go:523: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/426571b820df304bf0b1de52f751939f/logTestRegistryGC656873582
```

Jira issue: CRDB-34729

**Comment by adityamaru** [comment]
should be fixed by #116768

**Comment by stevendanna** [comment]
Should be fixed by merged PR.
